soccer/src/soccer/strategy/agent/position/offense.cpp

```diff
@@ -2,93 +2,161 @@
-    for (u_int32_t i = 0; i < response.responses.size(); i++) {
-        if (const communication::Acknowledge* acknowledge =
-                std::get_if<communication::Acknowledge>(&response.responses[i])) {
-            SPDLOG_INFO("\033[93mRobot {} has acknowledged the message: {}\033[0m",
-                        response.received_robot_ids[i]);
-        } else if (const communication::PassResponse* pass_response =
-                       std::get_if<communication::PassResponse>(&response.responses[i])) {
-            SPDLOG_INFO("\033[92mRobot {} has sent the pass response\033[0m",
-                        response.received_robot_ids[i]);
-        } else if (const communication::TestResponse* test_response =
-                       std::get_if<communication::TestResponse>(&response.responses[i])) {
-            SPDLOG_INFO("\033[91mRobot {} has sent a test response with message: {}\033[0m",
-                        response.received_robot_ids[i], test_response->message);
-        } else if (const communication::PositionResponse* position_response =
-                       std::get_if<communication::PositionResponse>(&response.responses[i])) {
-            SPDLOG_INFO("\033[90mRobot {} has sent a position response\033[0m",
-                        response.received_robot_ids[i]);
+std::optional<RobotIntent> Offense::state_to_task(RobotIntent intent) {
+    if (current_state_ == IDLING) {
+        // Do nothing
+        auto empty_motion_cmd = planning::MotionCommand{};
+        intent.motion_command = empty_motion_cmd;
+        return intent;
+    } else if (current_state_ == SEARCHING) {
+        // DEFINE SEARCHING BEHAVIOR
+    } else if (current_state_ == PASSING) {
+        // attempt to pass the ball to the target robot
+        rj_geometry::Point target_robot_pos =
+            world_state()->get_robot(true, target_robot_id).pose.position();
+        planning::LinearMotionInstant target{target_robot_pos};
+        auto line_kick_cmd = planning::MotionCommand{"line_kick", target};
+        intent.motion_command = line_kick_cmd;
+        intent.shoot_mode = RobotIntent::ShootMode::KICK;
+        // NOTE: Check we can actually use break beams
+        intent.trigger_mode = RobotIntent::TriggerMode::ON_BREAK_BEAM;
+        // TODO: Adjust the kick speed based on distance
+        intent.kick_speed = 4.0;
+        intent.is_active = true;
+        return intent;
+    } else if (current_state_ == SHOOTING) {
+        // TODO: Shoot the ball at the goal
+    } else if (current_state_ == RECEIVING) {
+        // check how far we are from the ball
+        rj_geometry::Point robot_position =
+            world_state()->get_robot(true, robot_id_).pose.position();
+        rj_geometry::Point ball_position = world_state()->ball.position;
+        double distance_to_ball = robot_position.dist_to(ball_position);
+        if (distance_to_ball > max_receive_distance && !chasing_ball) {
```

Comment:
what does this if check for? (add a comment or set a named boolean to clarify)
e.g. `bool too_far_to_intercept = ...`